apps/frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,75 @@
+import { type FC } from "react";
+
+import { KeyboardKey } from "~/libs/enums/enums.js";
+import {
+	type MouseEvent,
+	type KeyboardEvent as ReactKeyboardEvent,
+	type ReactNode,
+	useCallback,
+	useEffect,
+	useRef,
+} from "~/libs/hooks/hooks.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	children: ReactNode;
+	isOpen: boolean;
+	onClose: () => void;
+};
+
+const DEFAULT_TAB_INDEX = 0;
+
+const Modal: FC<Properties> = ({ children, isOpen, onClose }) => {
```

Comment:
Let's be consistent with other components
```suggestion
const Modal: React.FC<Properties> = ({ children, isOpen, onClose }) => {
```

Reply:
> Let's be consistent with other components

In this case, I get the warning: `'children' / 'isOpen' / 'onClose' is missing in props validation eslintreact/prop-types.`

Reply:
This approach should work
```suggestion
const Modal: React.FC<Properties> = ({ children, isOpen, onClose }: Properties) => {
```